core/src/test/java/com/coderanch/util/cli/InputUtilityTest.java

```diff
@@ -0,0 +1,126 @@
+package com.coderanch.util.cli;
+
+import junit.framework.TestCase;
+import org.junit.experimental.theories.Theories;
+import org.junit.experimental.theories.Theory;
+import org.junit.runner.RunWith;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+
+/**
+ * Tests the {@link InputUtility} class.
+ */
+@RunWith(Theories.class)
+public class InputUtilityTest {
+
+
+    /**
+     * @throws IOException
+     */
+    @Theory()
+    public void nextString_validatesCorrectInput() throws IOException {
+        var testCase = "input";
+        try(var s = new ByteArrayInputStream(testCase.getBytes())) {
```

Comment:
You can probably simplify a lot of code in this class if you add a utility method:

```java
private static <X> void withInput(
    String input,
    ExceptionalConsumer<? super InputUtility, ? extends X> action
) throws X {
    try (
        var stream = new ByteArrayInputStream(input.getBytes(StandardCharsets.UTF_8));
    ) {
        var inputUtility = new InputUtility(stream);
        action.accept(inputUtility);
    }
}
```
```java
@FunctionalInterface
private static interface ExceptionalConsumer<T, X> {

  void accept(T argument) throws X;
  
}
```
```java
withInput("input", inputUtility -> {
    var string = inputUtililty.nextString("prompt");
    assertThat("Next string must match expected value.", string, is("input"));
});
```